torchao/quantization/quant_primitives.py

```diff
@@ -1970,20 +1970,37 @@ def choose_qparams_affine_float8(
     tensor: torch.Tensor,
     float8_dtype: torch.dtype = torch.float8_e4m3fn,
     scale_dtype: torch.dtype = torch.float32,
+    block_size: Optional[Tuple[int, ...]] = None,
 ) -> torch.Tensor:
     """
     Calculates float8 scaling factor for the given high precision tensor, using tensorwise granularity.
 
     Args:
         tensor (torch.Tensor): Input tensor to be quantized.
         float8_dtype (torch.dtype): Data type of the quantized tensor (e.g., torch.float8_e4m3fn, torch.float8_e5m2).
+        scale_dtype (torch.dtype): Data type of the scaling factor (e.g., torch.float32).
+        block_size (Optional[Tuple[int, ...]]): Block size for block-wise quantization. If None, tensorwise quantization is used.
```

Comment:
was support for blockwise quantization intentionally included in this PR? this looks like the quant primitive support for it but presumably other things are still needed like scaling granularity (which only is PerTensor or PerRow right now), etc.

Reply:
This primitive, can/should be generic to block-size - which is what I did arguablly I hsould update the doc string a little, but its up to the caller to pass in the right block_size which in this case is only 2 difffernt forms today per-tensor and per-row